### PR TITLE
Handle gracefully if fs details can't be read

### PR DESCRIPTION
**Pull request by wolfs**
This seems to happen on Windows for network file systems.

### CODE REVIEW

**Comment by wolfs** [comment]
I tried out if the calls to `addUnknown` actually succeed by calling this method from native calls: https://builds.gradle.org/buildConfiguration/GradleNative_NativePlatform_BuildTrigger/42592390?buildTab=dependencies&mode=timeline&range=1618410033000-1618413963000

Seems to work.

File: native-platform/src/main/cpp/apple.cpp [hunk 1]
```diff
@@ -71,27 +72,27 @@ Java_net_rubygrapefruit_platform_internal_jni_PosixFileSystemFunctions_listFileS
...
-        }
-
-        if (alist.volattr & ATTR_VOL_CAPABILITIES) {
-            if ((buffer.caps.valid[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE)) {
-                caseSensitive = (buffer.caps.capabilities[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE) != 0;
+            env->CallVoidMethod(info, unknownFsMethod, mount_point, file_system_type, device_name, remote);
+        } else {
+            if (alist.volattr & ATTR_VOL_CAPABILITIES) {
+                if ((buffer.caps.valid[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE)) {
+                    caseSensitive = (buffer.caps.capabilities[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE) != 0;
```
Comment (lptr) [inline native-platform/src/main/cpp/apple.cpp:87]: Nt: let's move the definition of `caseSensitive` and `casePreserving` here.

File: native-platform/src/main/java/net/rubygrapefruit/platform/file/FileSystemInfo.java [hunk 2]
```diff
@@ -37,6 +37,20 @@
...
+     * You must not query additional details if they are unknown.
+     * <p>
+     * Additional details are:
+     * <ul>
+     *     <li>{@link #isCasePreserving()}</li>
+     *     <li>{@link #isCaseSensitive()}</li>
+     * </ul>
+     */
+    @ThreadSafe
+    boolean isDetailsKnown();
```
Comment (lptr) [inline native-platform/src/main/java/net/rubygrapefruit/platform/file/FileSystemInfo.java:52]: As this is part of the public API, I think it's important to have nice names, and `isDetailsKnown()` has a singular/plural collision. Maybe `isCaseSensitivityKnown()` would be a way to work around it?

**Review by lptr — COMMENTED** [review]
I'm a bit uneasy about the resulting API throwing exceptions when you access the `FileSystemInfo` instance.

I'd rather keep the current API fail, and introduce a new API to query file systems in a way that details are returned as `Optional` or something. WDYT?

File: native-platform/src/main/cpp/apple.cpp (hunk 1 above)
Comment (wolfs) [reply native-platform/src/main/cpp/apple.cpp:87]: Done.

**Comment by wolfs** [comment]
@lptr I changed how we handle the case when we can't determine the case sensitivity. The API is now a little bit different, though I think it is better that way. We can't use `Optional`, since we are still on Java 6, so I used a nullable method instead.

**Review by lptr — APPROVED** [review]
I like this solution, LGTM!

**Pull request merged**